packages/compartment-mapper/src/policy.js

```diff
@@ -381,57 +381,51 @@ export const enforceModulePolicy = (specifier, compartmentDescriptor, info) => {
  * @param {object} options
  * @param {DeferredAttenuatorsProvider} options.attenuators
  * @param {AttenuationDefinition} options.attenuationDefinition
- * @param {ModuleExportsNamespace} options.originalModule
- * @returns {ModuleExportsNamespace}
+ * @param {ThirdPartyStaticModuleInterface} options.originalModuleRecord
+ * @returns {ThirdPartyStaticModuleInterface}
  */
 function attenuateModule({
   attenuators,
   attenuationDefinition,
-  originalModule,
+  originalModuleRecord,
 }) {
-  const attenuationCompartment = new Compartment(
-    {},
-    {},
-    {
-      resolveHook: moduleSpecifier => moduleSpecifier,
-      importHook: async () => {
-        const attenuate = await importAttenuatorForDefinition(
-          attenuationDefinition,
-          attenuators,
-          MODULE_ATTENUATOR,
-        );
-        const ns = await attenuate(originalModule);
-        const staticModuleRecord = freeze({
-          imports: [],
-          exports: keys(ns),
-          execute: moduleExports => {
-            assign(moduleExports, ns);
-          },
-        });
-        return staticModuleRecord;
-      },
+  // No need for a new compartment anymore, since instead of a module namespace, we're now producing a module record.
```

Comment:
No need for a new compartment for each attenuation, since instead of a module namespace, we're now producing a module record!